Http/Middleware/RenderBlock.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Modules\Block\Http\Middleware;
+
+use Illuminate\Http\Response;
+use Modules\Block\Repositories\BlockRepository;
+
+class RenderBlock
+{
+    /** @var BlockRepository */
+    private $blockRepository;
+
+    public function __construct(BlockRepository $blockRepository)
+    {
+        $this->blockRepository = $blockRepository;
+    }
+
+    public function handle($request, \Closure $next)
+    {
+        /** @var Response $response */
+        $response = $next($request);
+
+        // do not render galleries on backend
+        if (app('asgard.onBackend') === true) {
+            return $response;
+        }
+
+        // if this is not a standard Response, return right away
+        if(!$response instanceof Response) {
+            return $response;
+        }
+
+        $response->setContent($this->replaceShortcodes($response->getContent()));
+
+        return $response;
+    }
+
+    /**
+     * replaces all block shortcodes from the response HTML with the actual block body
+     * @param string $html
+     * @return string
+     */
+    private function replaceShortcodes($html)
+    {
+        preg_match_all('/\[\[BLOCK\((.*)\)\]\]/U', $html, $matches);
+        $replaceBlocks = [];
+        foreach ($matches[1] as $blockIndex => $blockName) {
+            // prevent loading same block twice
+            if (isset($replaceBlocks[$matches[0][$blockIndex]])) {
+                continue;
+            }
+
+            $replaceBlocks[$matches[0][$blockIndex]] = $this->blockRepository->get($blockName);
```

Comment:
What happens if the block is not found?

Reply:
if the block is not found you mean if there is a shortcode, but the actual block does not exist in database ? Then the shortcode will be replaced by an empty string (result of the `BlockRepository@get` method - where I will also fix the phpdoc)

Reply:
I could also leave the actual shortcode in place if it is not found ... I don't have a preference here and I am happy to update the logic as you see fit